src/auth.rs

```diff
@@ -470,7 +461,78 @@ pub async fn verify_identity(
         // Keys server should never do this since it already returned SUCCESS above
         return Err(AuthError::KeyserverResponseMissingValue)?;
     };
-    let cacao = cacao.cacao;
+
+    Ok(cacao.cacao)
+}
+
+async fn keys_server_request_cached(
+    url: Url,
+    redis: Option<&Arc<Redis>>,
+) -> Result<(Cacao, KeysServerResponseSource)> {
+    let cache_key = format!("keys-server-{}", url);
+
+    if let Some(redis) = redis {
+        let value = redis.get(&cache_key).await?;
+        if let Some(cacao) = value {
+            return Ok((cacao, KeysServerResponseSource::Cache));
+        }
+    }
+
+    let cacao = keys_server_request(url).await?;
+
+    if let Some(redis) = redis {
+        let cacao = cacao.clone();
+        let redis = redis.clone();
+        let cache_ttl = chrono::Duration::weeks(1)
```

Comment:
Is it ok to have such a huge TTL for a cache?

Reply:
I think so, but am confirming with team https://walletconnect.slack.com/archives/C044SKFKELR/p1701995841390399